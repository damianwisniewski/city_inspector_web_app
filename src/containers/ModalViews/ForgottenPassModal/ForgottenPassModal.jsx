import React, { Component } from 'react'

// SCSS
import './ForgottenPassModal.scss'

// Components
import { Form, Input } from '../../../components/FormComponents'
import { Button, Status, RequestStatus } from '../../../components/CommonComponents'

// Others
import { Requester } from '../../../services/requester/requester'

class ForgottenPassModal extends Component {
	state = {
		sendRequestStatus: 'initial',
		email: '',
	}

	/**
	 * Sends requests for initialize remains password procedure
	 * User will receive mail with reset password link
	 * @param {Event} e
	 */
	handleRemainPass = e => {
		e.preventDefault()

		this.setState({ sendRequestStatus: 'pending' })

		Requester.send('initResetPassword', { body: { email: this.state.email } })
			.then(() => {
				this.setState({ sendRequestStatus: 'succeeded' })
			})
			.catch(() => {
				this.setState({ sendRequestStatus: 'failed' })
			})
	}

	/**
	 * Handler for input changes.
	 * Sets their values in state
	 * @param {Event} e
	 */
	handleInputChanges = e => {
		this.setState({
			email: e.target.value,
		})
	}

	render() {
		const { closeModal } = this.props
		const { email, sendRequestStatus } = this.state

		return (
			<div id='reminder-Modal' className='reminder'>
				<h2 className='reminder__header'>Resetowanie hasła</h2>
				<p className='reminder__description'>
					Podaj adres e-mail na który zarejestrowane jest konto
				</p>
				<Form onSubmit={this.handleRemainPass}>
					<Input
						id='reminder-email-field'
						onChange={this.handleInputChanges}
						value={email}
						label='e-mail'
						placeholder='Wpisz tutaj swój email...'
						labelType='floating'
						type='email'
						autocomplete='email'
						pattern='.+@.+'
						required
					/>
					<RequestStatus size='medium' requestState={sendRequestStatus}>
						<Status
							id='reminder-status'
							type='correct'
							position='vertical'
							message='Jeśli podałeś poprawny email, przypomnienie zostało wysłane'
						/>
					</RequestStatus>
					<div className='reminder__buttons-wrapper'>
						<Button id='reminder-close-button' color='white' onClick={closeModal}>
							Anuluj
						</Button>
						<Button id='reminder-submit-button' type='submit' color='blue'>
							Resetuj
						</Button>
					</div>
				</Form>
			</div>
		)
	}
}

export default ForgottenPassModal
