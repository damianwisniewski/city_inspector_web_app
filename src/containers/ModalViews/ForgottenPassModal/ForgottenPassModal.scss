@import '../../../assets/styleHelpers/index';

.reminder {
	width: 300px;
	height: 320px;
	font-size: 14px;
	text-align: center;
	position: relative;

	&__header {
		font-size: 20px;
		font-family: $secondary_font;
		color: $light_gray;
		background-color: $primary_blue;
		margin: 15px 10px;
		padding: 10px;
		border-radius: 5px;
		box-shadow: $inset-shadow;
		text-shadow: $text-shadow;
	}

	&__buttons-wrapper {
		margin-top: 20px;
		margin-bottom: 10px;
		position: absolute;
		bottom: 0;
		left: 0;
		width: 100%;
	}

	&__description {
		font-size: 15px;
		margin: 10px;
		color: $dark_gray;
	}
}
