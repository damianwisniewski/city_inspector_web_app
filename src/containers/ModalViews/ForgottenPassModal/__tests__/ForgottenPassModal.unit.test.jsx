import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import ForgottenPassModal from '../ForgottenPassModal'

configure({ adapter: new Adapter() })

describe('<ForgottenPassModal />', () => {
	test('should render correctly ForgottenPassModal', () => {
		const ForgottenPassModalElement = shallow(<ForgottenPassModal />)
		expect(toJson(ForgottenPassModalElement)).toMatchSnapshot()
	})
})
