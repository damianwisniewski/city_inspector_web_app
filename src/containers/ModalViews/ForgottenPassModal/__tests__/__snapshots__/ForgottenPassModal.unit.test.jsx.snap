// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgottenPassModal /> should render correctly ForgottenPassModal 1`] = `
<div
  className="reminder"
  id="reminder-Modal"
>
  <h2
    className="reminder__header"
  >
    Resetowanie hasła
  </h2>
  <p
    className="reminder__description"
  >
    Podaj adres e-mail na który zarejestrowane jest konto
  </p>
  <Form
    onSubmit={[Function]}
  >
    <Input
      autoComplete="email"
      id="reminder-email-field"
      inputRef={[Function]}
      label="e-mail"
      labelType="floating"
      onChange={[Function]}
      pattern=".+@.+"
      placeholder="Wpisz tutaj swój email..."
      required={true}
      type="email"
      value=""
    />
    <RequestStatus
      color="transparent"
      direction="horizontal"
      errorMessage="Przepraszamy coś poszło nie tak!"
      fullview={false}
      requestState="initial"
      size="medium"
    >
      <Status
        id="reminder-status"
        message="Jeśli podałeś poprawny email, przypomnienie zostało wysłane"
        position="vertical"
        type="correct"
      />
    </RequestStatus>
    <div
      className="reminder__buttons-wrapper"
    >
      <Button
        category="button"
        className=""
        color="white"
        id="reminder-close-button"
        onClick={[Function]}
        size="medium"
        type="button"
      >
        Anuluj
      </Button>
      <Button
        category="button"
        className=""
        color="blue"
        id="reminder-submit-button"
        onClick={[Function]}
        size="medium"
        type="submit"
      >
        Resetuj
      </Button>
    </div>
  </Form>
</div>
`;
